addons/delivery/models/stock_picking.py

```diff
@@ -92,6 +92,14 @@ def _compute_shipping_weight(self):
     package_ids = fields.Many2many('stock.quant.package', compute='_compute_packages', string='Packages')
     weight_bulk = fields.Float('Bulk Weight', compute='_compute_bulk_weight')
     shipping_weight = fields.Float("Weight for Shipping", compute='_compute_shipping_weight')
+    carrier_tracking_url = fields.Char(string='Tracking Url', compute='_compute_carrier_tracking_url', store=True)
+    is_sent_delivery_mail = fields.Boolean()
+
+    @api.depends('carrier_id', 'carrier_tracking_ref')
+    def _compute_carrier_tracking_url(self):
```

Comment:
carrier_tracking_url  generated when we have carrier and if user enter tracking ref manually then and then 
tracking url is generated. so that's why we put carrier_id and tracking ref on depends